dist/pythonlibs/riotctrl_ctrl/factory.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2021 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+
+import riotctrl.ctrl
+
+from .native import NativeRIOTCtrl
+
+
+class RIOTCtrlFactory:
+    """Abstract factory to create different RIOTCtrl types based on
+    the environment variables.
+
+    :param board_cls: A dict that maps a `BOARD` name to a RIOTCtrl class.
+    """
+    DEFAULT_CLS = riotctrl.ctrl.RIOTCtrl
+    BOARD_CLS = {
+        'native': NativeRIOTCtrl
+    }
```

Comment:
How would this evolve? Are we going to start implementing `BOARD_CLS` for all `BOARD`. Here I see the wrapper is only for setting the `DEBUG_ADAPTER_ID`. Do you foresee other functionalities for the `native` class?

Reply:
No, because that is mostly how the `native` board differs from other boards: We don't know the `DEBUG_ADAPTER_ID` before the instance starts, while for other boards it is an intrinsic property of the physical board.

Reply:
> No, because that is mostly how the `native` board differs from other boards: We don't know the `DEBUG_ADAPTER_ID` before the instance starts, while for other boards it is an intrinsic property of the physical board.

I like that argument.